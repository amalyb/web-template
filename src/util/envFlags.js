/**
 * Centralized environment flags for client-side code
 * Safely handles missing process object in browser environments
 */

export const IS_PROD = (typeof process !== 'undefined' && process.env && process.env.NODE_ENV === 'production');
export const IS_DEV  = (typeof process !== 'undefined' && process.env && process.env.NODE_ENV === 'development');
export const IS_TEST = (typeof process !== 'undefined' && process.env && process.env.NODE_ENV === 'test');
export const __DEV__ = !IS_PROD;
