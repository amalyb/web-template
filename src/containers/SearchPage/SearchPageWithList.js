import React, { Component } from 'react';
import { array, bool, func, oneOf, object, shape, string } from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { withRouter } from 'react-router-dom';
import omit from 'lodash/omit';
import classNames from 'classnames';

import config from '../../config';
import { injectIntl, intlShape, FormattedMessage } from '../../util/reactIntl';
import routeConfiguration from '../../routing/routeConfiguration';
import { createResourceLocatorString } from '../../util/routes';
import { isAnyFilterActive, isMainSearchTypeKeywords, getQueryParamNames } from '../../util/search';
import { parse, stringify } from '../../util/urlHelpers';
import { propTypes } from '../../util/types';
import { getListingsById } from '../../ducks/marketplaceData.duck';
import { manageDisableScrolling, isScrollingDisabled } from '../../ducks/UI.duck';

import { Footer, Page } from '../../components';

import TopbarContainer from '../../containers/TopbarContainer/TopbarContainer';

import {
  pickSearchParamsOnly,
  validURLParamsForExtendedData,
  validFilterParams,
  createSearchResultSchema,
} from './SearchPage.shared';

import FilterComponent from './FilterComponent';
import MainPanelHeader from './MainPanelHeader/MainPanelHeader';
import SearchFiltersMobile from './SearchFiltersMobile/SearchFiltersMobile';
import SortBy from './SortBy/SortBy';
import SearchResultsPanel from './SearchResultsPanel/SearchResultsPanel';
import NoSearchResultsMaybe from './NoSearchResultsMaybe/NoSearchResultsMaybe';

import css from './SearchPage.module.css';

const MODAL_BREAKPOINT = 768; // Search is in modal on mobile layout

// SortBy component has its content in dropdown-popup.
// With this offset we move the dropdown a few pixels on desktop layout.
const FILTER_DROPDOWN_OFFSET = -14;

const validUrlQueryParamsFromProps = props => {
  const { location, listingExtendedDataConfig, defaultFiltersConfig } = props;
  // eslint-disable-next-line no-unused-vars
  const { mapSearch, page, ...searchInURL } = parse(location.search, {
    latlng: ['origin'],
    latlngBounds: ['bounds'],
  });
  // urlQueryParams doesn't contain page specific url params
  // like mapSearch, page or origin (origin depends on config.sortSearchByDistance)
  return validURLParamsForExtendedData(
    searchInURL,
    listingExtendedDataConfig,
    defaultFiltersConfig
  );
};

const cleanSearchFromConflictingParams = (
  searchParams,
  listingExtendedDataConfig,
  defaultFiltersConfig,
  sortConfig
) => {
  // Single out filters that should disable SortBy when an active
  // keyword search sorts the listings according to relevance.
  // In those cases, sort parameter should be removed.
  const sortingFiltersActive = isAnyFilterActive(
    sortConfig.conflictingFilters,
    searchParams,
    listingExtendedDataConfig,
    defaultFiltersConfig
  );
  return sortingFiltersActive
    ? { ...searchParams, [sortConfig.queryParamName]: null }
    : searchParams;
};

export class SearchPageComponent extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isSearchMapOpenOnMobile: props.tab === 'map',
      isMobileModalOpen: false,
      currentQueryParams: validUrlQueryParamsFromProps(props),
    };

    this.searchMapListingsInProgress = false;

    this.onOpenMobileModal = this.onOpenMobileModal.bind(this);
    this.onCloseMobileModal = this.onCloseMobileModal.bind(this);

    // Filter functions
    this.resetAll = this.resetAll.bind(this);
    this.initialValues = this.initialValues.bind(this);
    this.getHandleChangedValueFn = this.getHandleChangedValueFn.bind(this);

    // SortBy
    this.handleSortBy = this.handleSortBy.bind(this);
  }

  // Invoked when a modal is opened from a child component,
  // for example when a filter modal is opened in mobile view
  onOpenMobileModal() {
    this.setState({ isMobileModalOpen: true });
  }

  // Invoked when a modal is closed from a child component,
  // for example when a filter modal is opened in mobile view
  onCloseMobileModal() {
    this.setState({ isMobileModalOpen: false });
  }

  // Reset all filter query parameters
  resetAll(e) {
    const { history, listingExtendedDataConfig, defaultFiltersConfig } = this.props;
    const urlQueryParams = validUrlQueryParamsFromProps(this.props);
    const filterQueryParamNames = getQueryParamNames(
      listingExtendedDataConfig,
      defaultFiltersConfig
    );

    // Reset state
    this.setState({ currentQueryParams: {} });

    // Reset routing params
    const queryParams = omit(urlQueryParams, filterQueryParamNames);
    history.push(createResourceLocatorString('SearchPage', routeConfiguration(), {}, queryParams));
  }

  initialValues(queryParamNames, isLiveEdit) {
    const urlQueryParams = validUrlQueryParamsFromProps(this.props);

    // Query parameters that are in state (user might have not yet clicked "Apply")
    const currentQueryParams = this.state.currentQueryParams;

    // Get initial value for a given parameter from state if its there.
    const getInitialValue = paramName => {
      const currentQueryParam = currentQueryParams[paramName];
      const hasQueryParamInState = typeof currentQueryParam !== 'undefined';
      return hasQueryParamInState && !isLiveEdit ? currentQueryParam : urlQueryParams[paramName];
    };

    // Return all the initial values related to given queryParamNames
    // InitialValues for "amenities" filter could be
    // { amenities: "has_any:towel,jacuzzi" }
    const isArray = Array.isArray(queryParamNames);
    return isArray
      ? queryParamNames.reduce((acc, paramName) => {
          return { ...acc, [paramName]: getInitialValue(paramName) };
        }, {})
      : {};
  }

  getHandleChangedValueFn(useHistoryPush) {
    const { history, sortConfig, listingExtendedDataConfig, defaultFiltersConfig } = this.props;
    const urlQueryParams = validUrlQueryParamsFromProps(this.props);

    return updatedURLParams => {
      const updater = prevState => {
        const { address, bounds, keywords } = urlQueryParams;
        const mergedQueryParams = { ...urlQueryParams, ...prevState.currentQueryParams };

        // Address and bounds are handled outside of MainPanel.
        // I.e. TopbarSearchForm && search by moving the map.
        // We should always trust urlQueryParams with those.
        // The same applies to keywords, if the main search type is keyword search.
        const keywordsMaybe = isMainSearchTypeKeywords(config) ? { keywords } : {};
        return {
          currentQueryParams: {
            ...mergedQueryParams,
            ...updatedURLParams,
            ...keywordsMaybe,
            address,
            bounds,
          },
        };
      };

      const callback = () => {
        if (useHistoryPush) {
          const searchParams = this.state.currentQueryParams;
          const search = cleanSearchFromConflictingParams(
            searchParams,
            listingExtendedDataConfig,
            defaultFiltersConfig,
            sortConfig
          );
          history.push(createResourceLocatorString('SearchPage', routeConfiguration(), {}, search));
        }
      };

      this.setState(updater, callback);
    };
  }

  handleSortBy(urlParam, values) {
    const { history } = this.props;
    const urlQueryParams = validUrlQueryParamsFromProps(this.props);

    const queryParams = values
      ? { ...urlQueryParams, [urlParam]: values }
      : omit(urlQueryParams, urlParam);

    history.push(createResourceLocatorString('SearchPage', routeConfiguration(), {}, queryParams));
  }

  // Reset all filter query parameters
  handleResetAll(e) {
    this.resetAll(e);

    // blur event target if event is passed
    if (e && e.currentTarget) {
      e.currentTarget.blur();
    }
  }

  render() {
    const {
      intl,
      listings,
      activeProcesses,
      listingExtendedDataConfig,
      defaultFiltersConfig,
      sortConfig,
      location,
      onManageDisableScrolling,
      pagination,
      scrollingDisabled,
      searchInProgress,
      searchListingsError,
      searchParams,
    } = this.props;
    // eslint-disable-next-line no-unused-vars
    const { mapSearch, page, ...searchInURL } = parse(location.search, {
      latlng: ['origin'],
      latlngBounds: ['bounds'],
    });

    // urlQueryParams doesn't contain page specific url params
    // like mapSearch, page or origin (origin depends on config.sortSearchByDistance)
    const urlQueryParams = pickSearchParamsOnly(
      searchInURL,
      listingExtendedDataConfig,
      defaultFiltersConfig,
      sortConfig
    );

    // Page transition might initially use values from previous search
    const urlQueryString = stringify(urlQueryParams);
    const paramsQueryString = stringify(
      pickSearchParamsOnly(
        searchParams,
        listingExtendedDataConfig,
        defaultFiltersConfig,
        sortConfig
      )
    );
    const searchParamsAreInSync = urlQueryString === paramsQueryString;

    const validQueryParams = validURLParamsForExtendedData(
      searchInURL,
      listingExtendedDataConfig,
      defaultFiltersConfig
    );

    const isKeywordSearch = isMainSearchTypeKeywords(config);
    const defaultFilters = isKeywordSearch
      ? defaultFiltersConfig.filter(f => f.key !== 'keywords')
      : defaultFiltersConfig;
    const groupConfigs = configs =>
      configs.reduce(
        (grouped, config) => {
          const [primary, secondary] = grouped;
          const { includeForProcessAliases, indexForSearch, searchPageConfig } = config;
          const isIndexed = indexForSearch === true;
          const isActiveProcess = includeForProcessAliases.every(p =>
            activeProcesses.includes(p.split('/')[0])
          );
          const isPrimary = searchPageConfig?.group === 'primary';
          return isActiveProcess && isIndexed && isPrimary
            ? [[...primary, config], secondary]
            : isActiveProcess && isIndexed
            ? [primary, [...secondary, config]]
            : grouped;
        },
        [[], []]
      );
    const [customPrimaryFilters, customSecondaryFilters] = groupConfigs(listingExtendedDataConfig);
    const availableFilters = [
      ...customPrimaryFilters,
      ...defaultFilters,
      ...customSecondaryFilters,
    ];

    // Selected aka active filters
    const selectedFilters = validFilterParams(
      validQueryParams,
      listingExtendedDataConfig,
      defaultFiltersConfig
    );
    const keysOfSelectedFilters = Object.keys(selectedFilters);
    const selectedFiltersCountForMobile = isKeywordSearch
      ? keysOfSelectedFilters.filter(f => f !== 'keywords').length
      : keysOfSelectedFilters.length;

    const hasPaginationInfo = !!pagination && pagination.totalItems != null;
    const totalItems =
      searchParamsAreInSync && hasPaginationInfo
        ? pagination.totalItems
        : pagination?.paginationUnsupported
        ? listings.length
        : 0;
    const listingsAreLoaded =
      !searchInProgress &&
      searchParamsAreInSync &&
      (hasPaginationInfo || pagination?.paginationUnsupported);

    const sortBy = mode => {
      const conflictingFilterActive = isAnyFilterActive(
        sortConfig.conflictingFilters,
        validQueryParams,
        listingExtendedDataConfig,
        defaultFiltersConfig
      );

      const mobileClassesMaybe =
        mode === 'mobile'
          ? {
              rootClassName: css.sortBy,
              menuLabelRootClassName: css.sortByMenuLabel,
            }
          : { className: css.sortByDesktop };
      return sortConfig.active ? (
        <SortBy
          {...mobileClassesMaybe}
          sort={validQueryParams[sortConfig.queryParamName]}
          isConflictingFilterActive={!!conflictingFilterActive}
          hasConflictingFilters={!!(sortConfig.conflictingFilters?.length > 0)}
          selectedFilters={selectedFilters}
          onSelect={this.handleSortBy}
          showAsPopup
          contentPlacementOffset={FILTER_DROPDOWN_OFFSET}
        />
      ) : null;
    };
    const noResultsInfo = (
      <NoSearchResultsMaybe
        listingsAreLoaded={listingsAreLoaded}
        totalItems={totalItems}
        location={location}
        resetAll={this.resetAll}
      />
    );

    const { title, description, schema } = createSearchResultSchema(
      listings,
      searchInURL || {},
      intl
    );

    // Set topbar class based on if a modal is open in
    // a child component
    const topbarClasses = this.state.isMobileModalOpen
      ? classNames(css.topbarBehindModal, css.topbar)
      : css.topbar;

    // N.B. openMobileMap button is sticky.
    // For some reason, stickyness doesn't work on Safari, if the element is <button>
    return (
      <Page
        scrollingDisabled={scrollingDisabled}
        description={description}
        title={title}
        schema={schema}
      >
        <TopbarContainer
          className={topbarClasses}
          currentPage="SearchPage"
          currentSearchParams={urlQueryParams}
        />
        <div className={css.layoutWrapperContainer}>
          <aside className={css.layoutWrapperFilterColumn}>
            <div className={css.filterColumnContent}>
              {availableFilters.map(config => {
                return (
                  <FilterComponent
                    key={`SearchFiltersMobile.${config.key}`}
                    idPrefix="SearchFiltersMobile"
                    className={css.filter}
                    config={config}
                    urlQueryParams={urlQueryParams}
                    initialValues={this.initialValues}
                    getHandleChangedValueFn={this.getHandleChangedValueFn}
                    liveEdit
                    showAsPopup={false}
                    isDesktop
                  />
                );
              })}
              <button className={css.resetAllButton} onClick={e => this.handleResetAll(e)}>
                <FormattedMessage id={'SearchFiltersMobile.resetAll'} />
              </button>
            </div>
          </aside>

          <div className={css.layoutWrapperMain} role="main">
            <div className={css.searchResultContainer}>
              <SearchFiltersMobile
                className={css.searchFiltersMobileList}
                urlQueryParams={validQueryParams}
                sortByComponent={sortBy('mobile')}
                listingsAreLoaded={listingsAreLoaded}
                resultsCount={totalItems}
                searchInProgress={searchInProgress}
                searchListingsError={searchListingsError}
                showAsModalMaxWidth={MODAL_BREAKPOINT}
                onManageDisableScrolling={onManageDisableScrolling}
                onOpenModal={this.onOpenMobileModal}
                onCloseModal={this.onCloseMobileModal}
                resetAll={this.resetAll}
                selectedFiltersCount={selectedFiltersCountForMobile}
                isMapVariant={false}
                noResultsInfo={noResultsInfo}
              >
                {availableFilters.map(config => {
                  return (
                    <FilterComponent
                      key={`SearchFiltersMobile.${config.key}`}
                      idPrefix="SearchFiltersMobile"
                      config={config}
                      urlQueryParams={validQueryParams}
                      initialValues={this.initialValues}
                      getHandleChangedValueFn={this.getHandleChangedValueFn}
                      liveEdit
                      showAsPopup={false}
                    />
                  );
                })}
              </SearchFiltersMobile>
              <MainPanelHeader
                className={css.mainPanel}
                sortByComponent={sortBy('desktop')}
                listingsAreLoaded={listingsAreLoaded}
                resultsCount={totalItems}
                searchInProgress={searchInProgress}
                searchListingsError={searchListingsError}
                noResultsInfo={noResultsInfo}
              />
              <div
                className={classNames(css.listings, {
                  [css.newSearchInProgress]: !listingsAreLoaded,
                })}
              >
                {searchListingsError ? (
                  <h2 className={css.error}>
                    <FormattedMessage id="SearchPage.searchError" />
                  </h2>
                ) : null}
                <SearchResultsPanel
                  className={css.searchListingsPanel}
                  listings={listings}
                  pagination={listingsAreLoaded ? pagination : null}
                  search={parse(location.search)}
                  isMapVariant={false}
                />
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </Page>
    );
  }
}

SearchPageComponent.defaultProps = {
  listings: [],
  pagination: null,
  searchListingsError: null,
  searchParams: {},
  tab: 'listings',
  activeProcesses: config.custom.processes,
  listingExtendedDataConfig: config.custom.listingExtendedData,
  defaultFiltersConfig: config.custom.defaultFilters,
  sortConfig: config.custom.sortConfig,
};

SearchPageComponent.propTypes = {
  listings: array,
  onManageDisableScrolling: func.isRequired,
  pagination: propTypes.pagination,
  scrollingDisabled: bool.isRequired,
  searchInProgress: bool.isRequired,
  searchListingsError: propTypes.error,
  searchParams: object,
  tab: oneOf(['filters', 'listings', 'map']).isRequired,
  activeProcesses: array,
  listingExtendedDataConfig: propTypes.listingExtendedDataConfig,
  defaultFiltersConfig: propTypes.defaultFiltersConfig,
  sortConfig: propTypes.sortConfig,

  // from withRouter
  history: shape({
    push: func.isRequired,
  }).isRequired,
  location: shape({
    search: string.isRequired,
  }).isRequired,

  // from injectIntl
  intl: intlShape.isRequired,
};

const mapStateToProps = state => {
  const {
    currentPageResultIds,
    pagination,
    searchInProgress,
    searchListingsError,
    searchParams,
  } = state.SearchPage;
  const listings = getListingsById(state, currentPageResultIds);

  return {
    listings,
    pagination,
    scrollingDisabled: isScrollingDisabled(state),
    searchInProgress,
    searchListingsError,
    searchParams,
  };
};

const mapDispatchToProps = dispatch => ({
  onManageDisableScrolling: (componentId, disableScrolling) =>
    dispatch(manageDisableScrolling(componentId, disableScrolling)),
});

// Note: it is important that the withRouter HOC is **outside** the
// connect HOC, otherwise React Router won't rerender any Route
// components since connect implements a shouldComponentUpdate
// lifecycle hook.
//
// See: https://github.com/ReactTraining/react-router/issues/4671
const SearchPage = compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  injectIntl
)(SearchPageComponent);

export default SearchPage;
