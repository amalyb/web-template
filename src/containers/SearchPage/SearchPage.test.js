import React from 'react';
import { getRouteConfiguration, renderShallow } from '../../util/test-helpers';
import { fakeIntl } from '../../util/test-data';
import { SearchPageComponent } from './SearchPageWithList';

const noop = () => null;

const transactionTypes = [
  {
    type: 'rent-bicycles-daily',
    process: 'default-booking',
    alias: 'release-1',
    unitType: 'day',
  },
  {
    type: 'rent-bicycles-nightly',
    process: 'default-booking',
    alias: 'release-1',
    unitType: 'night',
  },
  {
    type: 'rent-bicycles-hourly',
    process: 'default-booking',
    alias: 'release-1',
    unitType: 'hour',
  },
  {
    type: 'sell-bicycles',
    process: 'default-buying-products',
    alias: 'release-1',
    unitType: 'item',
  },
];

const listingExtendedDataConfig = [
  {
    key: 'category',
    scope: 'public',
    includeForTransactionTypes: ['sell-bicycles'],
    schemaType: 'enum',
    schemaOptions: [{ option: 'cat_1', label: 'Cat 1' }, { option: 'cat_2', label: 'Cat 2' }],
    indexForSearch: true,
    searchPageConfig: {
      label: 'Category',
      group: 'primary',
    },
    listingPageConfig: {
      label: 'Category',
    },
    editListingPageConfig: {
      label: 'Category',
    },
  },
  {
    key: 'amenities',
    scope: 'public',
    includeForTransactionTypes: [
      'rent-bicycles-daily',
      'rent-bicycles-nightly',
      'rent-bicycles-hourly',
    ],
    schemaType: 'multi-enum',
    schemaOptions: [{ option: 'dog_1', label: 'Dog 1' }, { option: 'dog_2', label: 'Dog 2' }],
    indexForSearch: true,
    searchPageConfig: {
      label: 'Amenities',
      //searchMode: 'has_all',
      group: 'secondary',
    },
    listingPageConfig: {
      label: 'Category',
    },
    editListingPageConfig: {
      label: 'Category',
    },
  },
];

const defaultFiltersConfig = [
  {
    key: 'price',
    schemaType: 'price',
    label: 'Price',
    // Note: unlike most prices this is not handled in subunits
    min: 0,
    max: 1000,
    step: 5,
  },
  {
    key: 'keywords',
    schemaType: 'text',
    label: 'Keyword',
  },
];

const sortConfig = {
  active: true,
  queryParamName: 'sort',
  relevanceKey: 'relevance',
  conflictingFilters: [],
  options: [
    { key: 'createdAt', label: 'Newest' },
    { key: '-createdAt', label: 'Oldest' },
    { key: '-price', label: 'Lowest price' },
    { key: 'price', label: 'Highest price' },
    { key: 'relevance', label: 'Relevance', longLabel: 'Relevance (Keyword search)' },
  ],
};

describe('SearchPageWithList', () => {
  it('matches snapshot', () => {
    const props = {
      location: { search: '' },
      history: {
        push: () => console.log('HistoryPush called'),
      },
      pagination: {
        page: 1,
        perPage: 12,
        totalItems: 22,
        totalPages: 2,
      },
      tab: 'listings',
      scrollingDisabled: false,
      searchInProgress: false,
      authInProgress: false,
      currentUserHasListings: false,
      intl: fakeIntl,
      isAuthenticated: false,
      onActivateListing: noop,
      onLogout: noop,
      onManageDisableScrolling: noop,
      onSearchMapListings: noop,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
      config: {
        listing: {
          listingExtendedData: listingExtendedDataConfig,
        },
        search: {
          mainSearchType: 'location',
          defaultFilters: defaultFiltersConfig,
          sortConfig: sortConfig,
        },
        transaction: {
          transactionTypes,
        },
        maps: {
          search: {
            sortSearchByDistance: false,
          },
        },
        layout: {
          searchPageVariant: 'list',
        },
      },
      routeConfiguration: getRouteConfiguration(),
    };
    const tree = renderShallow(<SearchPageComponent {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
