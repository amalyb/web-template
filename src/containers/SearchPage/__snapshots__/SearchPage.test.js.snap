// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPageWithList matches snapshot 1`] = `
<Page
  description="SearchPage.schemaDescription"
  schema={
    Object {
      "@context": "http://schema.org",
      "@type": "SearchResultsPage",
      "description": "SearchPage.schemaDescription",
      "mainEntity": Array [
        "{\\"@type\\":\\"ItemList\\",\\"name\\":\\"SearchPage.schemaForSearch\\",\\"itemListOrder\\":\\"http://schema.org/ItemListOrderAscending\\",\\"itemListElement\\":[]}",
      ],
      "name": "SearchPage.schemaTitle",
    }
  }
  scrollingDisabled={false}
  title="SearchPage.schemaTitle"
>
  <withRouter(Connect(TopbarContainerComponent))
    className="topbar"
    currentPage="SearchPage"
    currentSearchParams={Object {}}
  />
  <div
    className="layoutWrapperContainer"
  >
    <aside
      className="layoutWrapperFilterColumn"
    >
      <div
        className="filterColumnContent"
      >
        <FilterComponent
          className="filter"
          config={
            Object {
              "editListingPageConfig": Object {
                "label": "Category",
              },
              "includeForProcessAliases": Array [
                "flex-product-default-process/release-1",
              ],
              "indexForSearch": true,
              "key": "category",
              "listingPageConfig": Object {
                "label": "Category",
              },
              "schemaOptions": Array [
                Object {
                  "label": "Cat 1",
                  "option": "cat_1",
                },
                Object {
                  "label": "Cat 2",
                  "option": "cat_2",
                },
              ],
              "schemaType": "enum",
              "scope": "public",
              "searchPageConfig": Object {
                "group": "primary",
                "label": "Category",
              },
            }
          }
          getHandleChangedValueFn={[Function]}
          idPrefix="SearchFiltersMobile"
          initialValues={[Function]}
          intl={
            Object {
              "formatDate": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelativeTime": [Function],
              "formatTime": [Function],
              "now": [Function],
            }
          }
          isDesktop={true}
          key="SearchFiltersMobile.category"
          liveEdit={true}
          showAsPopup={false}
          urlQueryParams={Object {}}
        />
        <FilterComponent
          className="filter"
          config={
            Object {
              "key": "price",
              "label": "Price",
              "max": 1000,
              "min": 0,
              "schemaType": "price",
              "step": 5,
            }
          }
          getHandleChangedValueFn={[Function]}
          idPrefix="SearchFiltersMobile"
          initialValues={[Function]}
          intl={
            Object {
              "formatDate": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelativeTime": [Function],
              "formatTime": [Function],
              "now": [Function],
            }
          }
          isDesktop={true}
          key="SearchFiltersMobile.price"
          liveEdit={true}
          showAsPopup={false}
          urlQueryParams={Object {}}
        />
        <FilterComponent
          className="filter"
          config={
            Object {
              "key": "keywords",
              "label": "Keyword",
              "schemaType": "text",
            }
          }
          getHandleChangedValueFn={[Function]}
          idPrefix="SearchFiltersMobile"
          initialValues={[Function]}
          intl={
            Object {
              "formatDate": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelativeTime": [Function],
              "formatTime": [Function],
              "now": [Function],
            }
          }
          isDesktop={true}
          key="SearchFiltersMobile.keywords"
          liveEdit={true}
          showAsPopup={false}
          urlQueryParams={Object {}}
        />
        <FilterComponent
          className="filter"
          config={
            Object {
              "editListingPageConfig": Object {
                "label": "Category",
              },
              "includeForProcessAliases": Array [
                "flex-booking-default-process/release-1",
              ],
              "indexForSearch": true,
              "key": "amenities",
              "listingPageConfig": Object {
                "label": "Category",
              },
              "schemaOptions": Array [
                Object {
                  "label": "Dog 1",
                  "option": "dog_1",
                },
                Object {
                  "label": "Dog 2",
                  "option": "dog_2",
                },
              ],
              "schemaType": "multi-enum",
              "scope": "public",
              "searchPageConfig": Object {
                "group": "secondary",
                "label": "Amenities",
              },
            }
          }
          getHandleChangedValueFn={[Function]}
          idPrefix="SearchFiltersMobile"
          initialValues={[Function]}
          intl={
            Object {
              "formatDate": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatPlural": [Function],
              "formatRelativeTime": [Function],
              "formatTime": [Function],
              "now": [Function],
            }
          }
          isDesktop={true}
          key="SearchFiltersMobile.amenities"
          liveEdit={true}
          showAsPopup={false}
          urlQueryParams={Object {}}
        />
        <button
          className="resetAllButton"
          onClick={[Function]}
        >
          <MemoizedFormattedMessage
            id="SearchFiltersMobile.resetAll"
          />
        </button>
      </div>
    </aside>
    <div
      className="layoutWrapperMain"
      role="main"
    >
      <div
        className="searchResultContainer"
      >
        <SearchFiltersMobile
          className="searchFiltersMobileList"
          isMapVariant={false}
          listingsAreLoaded={true}
          noResultsInfo={
            <NoSearchResultsMaybe
              listingsAreLoaded={true}
              location={
                Object {
                  "search": "",
                }
              }
              resetAll={[Function]}
              totalItems={22}
            />
          }
          onCloseModal={[Function]}
          onManageDisableScrolling={[Function]}
          onOpenModal={[Function]}
          resetAll={[Function]}
          resultsCount={22}
          searchInProgress={false}
          searchListingsError={null}
          selectedFiltersCount={0}
          showAsModalMaxWidth={768}
          sortByComponent={
            <injectIntl(SortBy)
              contentPlacementOffset={-14}
              hasConflictingFilters={false}
              isConflictingFilterActive={false}
              mode="mobile"
              onSelect={[Function]}
              selectedFilters={Object {}}
              showAsPopup={true}
            />
          }
          urlQueryParams={Object {}}
        >
          <FilterComponent
            config={
              Object {
                "editListingPageConfig": Object {
                  "label": "Category",
                },
                "includeForProcessAliases": Array [
                  "flex-product-default-process/release-1",
                ],
                "indexForSearch": true,
                "key": "category",
                "listingPageConfig": Object {
                  "label": "Category",
                },
                "schemaOptions": Array [
                  Object {
                    "label": "Cat 1",
                    "option": "cat_1",
                  },
                  Object {
                    "label": "Cat 2",
                    "option": "cat_2",
                  },
                ],
                "schemaType": "enum",
                "scope": "public",
                "searchPageConfig": Object {
                  "group": "primary",
                  "label": "Category",
                },
              }
            }
            getHandleChangedValueFn={[Function]}
            idPrefix="SearchFiltersMobile"
            initialValues={[Function]}
            intl={
              Object {
                "formatDate": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            key="SearchFiltersMobile.category"
            liveEdit={true}
            showAsPopup={false}
            urlQueryParams={Object {}}
          />
          <FilterComponent
            config={
              Object {
                "key": "price",
                "label": "Price",
                "max": 1000,
                "min": 0,
                "schemaType": "price",
                "step": 5,
              }
            }
            getHandleChangedValueFn={[Function]}
            idPrefix="SearchFiltersMobile"
            initialValues={[Function]}
            intl={
              Object {
                "formatDate": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            key="SearchFiltersMobile.price"
            liveEdit={true}
            showAsPopup={false}
            urlQueryParams={Object {}}
          />
          <FilterComponent
            config={
              Object {
                "key": "keywords",
                "label": "Keyword",
                "schemaType": "text",
              }
            }
            getHandleChangedValueFn={[Function]}
            idPrefix="SearchFiltersMobile"
            initialValues={[Function]}
            intl={
              Object {
                "formatDate": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            key="SearchFiltersMobile.keywords"
            liveEdit={true}
            showAsPopup={false}
            urlQueryParams={Object {}}
          />
          <FilterComponent
            config={
              Object {
                "editListingPageConfig": Object {
                  "label": "Category",
                },
                "includeForProcessAliases": Array [
                  "flex-booking-default-process/release-1",
                ],
                "indexForSearch": true,
                "key": "amenities",
                "listingPageConfig": Object {
                  "label": "Category",
                },
                "schemaOptions": Array [
                  Object {
                    "label": "Dog 1",
                    "option": "dog_1",
                  },
                  Object {
                    "label": "Dog 2",
                    "option": "dog_2",
                  },
                ],
                "schemaType": "multi-enum",
                "scope": "public",
                "searchPageConfig": Object {
                  "group": "secondary",
                  "label": "Amenities",
                },
              }
            }
            getHandleChangedValueFn={[Function]}
            idPrefix="SearchFiltersMobile"
            initialValues={[Function]}
            intl={
              Object {
                "formatDate": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            key="SearchFiltersMobile.amenities"
            liveEdit={true}
            showAsPopup={false}
            urlQueryParams={Object {}}
          />
        </SearchFiltersMobile>
        <MainPanelHeader
          className="mainPanel"
          isSortByActive={true}
          listingsAreLoaded={true}
          noResultsInfo={
            <NoSearchResultsMaybe
              listingsAreLoaded={true}
              location={
                Object {
                  "search": "",
                }
              }
              resetAll={[Function]}
              totalItems={22}
            />
          }
          resultsCount={22}
          rootClassName={null}
          searchInProgress={false}
          searchListingsError={null}
          sortByComponent={
            <injectIntl(SortBy)
              contentPlacementOffset={-14}
              hasConflictingFilters={false}
              isConflictingFilterActive={false}
              mode="desktop"
              onSelect={[Function]}
              selectedFilters={Object {}}
              showAsPopup={true}
            />
          }
        />
        <div
          className="listings"
        >
          <SearchResultsPanel
            className="searchListingsPanel"
            isMapVariant={false}
            listings={Array []}
            pagination={
              Object {
                "page": 1,
                "perPage": 12,
                "totalItems": 22,
                "totalPages": 2,
              }
            }
            rootClassName={null}
            search={Object {}}
          />
        </div>
      </div>
    </div>
  </div>
  <injectIntl(Footer) />
</Page>
`;
